test/examples/regression/issue328_dependent_name/issue328_dependent_name.py

```diff
@@ -0,0 +1,18 @@
+import unittest
+
+import issue328_nested_class_name_pybind11 as py11
+import issue328_nested_class_name_boost_python as boost
+
+
+class TestIssue328(unittest.TestCase):
+
+    def _test_issue328(self, binding):
+        pass
+
+    def test_issue328(self):
+        self._test_issue328(py11)
+        self._test_issue328(boost)
```

Comment:
In situations where we are just suppressing an issue instead of solving it, could we add an assertion that _should_ work if we didn't have the error (like being able to access this `::type`) and then comment it out in the test?

I am worried it will be hard to tell which issues we have solved and which ones we have just covered up, which will make it hard if we e.g. upgrade clang then some of this functionality is fixed.